### PR TITLE
JsonSchema should not validate the $schema field

### PR DESCRIPTION
**Pull request by nickpape-msft**

### CODE REVIEW

File: libraries/node-core-library/src/JsonSchema.ts [hunk 1]
```diff
@@ -271,6 +271,9 @@ export class JsonSchema {
 
     this.ensureCompiled();
 
+    // remove the $schema field
+    delete jsonObject['$schema'];
```
Comment (pgonzal) [inline libraries/node-core-library/src/JsonSchema.ts]: Don't do that.

See [https://github.com/Microsoft/web-build-tools/pull/341](https://github.com/Microsoft/web-build-tools/pull/341)

Comment (pgonzal) [reply libraries/node-core-library/src/JsonSchema.ts]: Due to the simpleminded design of JSON schemas, the $schema metadata field needs to be explicitly declared as part of each schema.  Various schema validators such as VS Code will highlight it as an error otherwise.

In the past, our schema parsing utility did delete this field, but that was a workaround because we didn't have URLs allocated (so we couldn't easily map them).  When we fixed that, our validator became stricter.

In short:  You need to fix your schema.  Nothing is wrong with the validator.
